Wallet of owner tests fail

Hi,

After running: 
> npx hardhat test

right after cloning this repository the "Wallet of owner tests" fail.

This is the error message I get:
`AssertionError: expected [ BigNumber{ _hex: '0x01', …(1) } ] to deeply equal [ BigNumber { value: "1" } ] + expected - actual at Context.<anonymous> (test/index.ts:205:73)`

What is going wrong here?

--- a/smart-contract/contracts/YourNftToken.sol
+++ b/smart-contract/contracts/YourNftToken.sol
@@ -77,18 +77,12 @@ contract YourNftToken is ERC721A, Ownable, ReentrancyGuard {
     uint256[] memory ownedTokenIds = new uint256[](ownerTokenCount);
     uint256 currentTokenId = _startTokenId();
     uint256 ownedTokenIndex = 0;
-    address latestOwnerAddress;
 
     while (ownedTokenIndex < ownerTokenCount && currentTokenId <= maxSupply) {
-      if (_exists(currentTokenId))
-      {
+      if (_exists(currentTokenId)) {
         TokenOwnership memory ownership = ownershipOf(currentTokenId);
 
-        if (!ownership.burned && ownership.addr != address(0)) {
-          latestOwnerAddress = ownership.addr;
-        }
-
-        if (latestOwnerAddress == _owner) {
+        if (ownership.addr == _owner) {
           ownedTokenIds[ownedTokenIndex] = currentTokenId;
 
           ownedTokenIndex++;

--- a/smart-contract/contracts/YourNftToken.sol
+++ b/smart-contract/contracts/YourNftToken.sol
@@ -77,18 +77,23 @@ contract YourNftToken is ERC721A, Ownable, ReentrancyGuard {
     uint256[] memory ownedTokenIds = new uint256[](ownerTokenCount);
     uint256 currentTokenId = _startTokenId();
     uint256 ownedTokenIndex = 0;
+    address latestOwnerAddress;
 
     while (ownedTokenIndex < ownerTokenCount && currentTokenId <= maxSupply) {
       if (_exists(currentTokenId)) {
-        TokenOwnership memory ownership = ownershipOf(currentTokenId);
+        TokenOwnership memory ownership = _ownerships[currentTokenId];
 
-        if (ownership.addr == _owner) {
+        if (ownership.addr != address(0)) {
+          latestOwnerAddress = ownership.addr;
+        }
+
+        if (latestOwnerAddress == _owner) {
           ownedTokenIds[ownedTokenIndex] = currentTokenId;
 
           ownedTokenIndex++;
         }
       }
-      
+
       currentTokenId++;
     }
 

--- a/smart-contract/contracts/YourNftToken.sol
+++ b/smart-contract/contracts/YourNftToken.sol
@@ -93,9 +93,8 @@ contract YourNftToken is ERC721A, Ownable, ReentrancyGuard {
 
           ownedTokenIndex++;
         }
-
-        currentTokenId++;
       }
+      currentTokenId++;
     }
 
     return ownedTokenIds;

--- a/smart-contract/contracts/YourNftToken.sol
+++ b/smart-contract/contracts/YourNftToken.sol
@@ -80,19 +80,22 @@ contract YourNftToken is ERC721A, Ownable, ReentrancyGuard {
     address latestOwnerAddress;
 
     while (ownedTokenIndex < ownerTokenCount && currentTokenId <= maxSupply) {
-      TokenOwnership memory ownership = _ownerships[currentTokenId];
+      if (_exists(currentTokenId))
+      {
+        TokenOwnership memory ownership = ownershipOf(currentTokenId);
 
-      if (!ownership.burned && ownership.addr != address(0)) {
-        latestOwnerAddress = ownership.addr;
+        if (!ownership.burned && ownership.addr != address(0)) {
+          latestOwnerAddress = ownership.addr;
+        }
+
+        if (latestOwnerAddress == _owner) {
+          ownedTokenIds[ownedTokenIndex] = currentTokenId;
+
+          ownedTokenIndex++;
+        }
+
+        currentTokenId++;
       }
-
-      if (latestOwnerAddress == _owner) {
-        ownedTokenIds[ownedTokenIndex] = currentTokenId;
-
-        ownedTokenIndex++;
-      }
-
-      currentTokenId++;
     }
 
     return ownedTokenIds;

--- a/smart-contract/contracts/YourNftToken.sol
+++ b/smart-contract/contracts/YourNftToken.sol
@@ -88,6 +88,7 @@ contract YourNftToken is ERC721A, Ownable, ReentrancyGuard {
           ownedTokenIndex++;
         }
       }
+      
       currentTokenId++;
     }
 

